node/src/signature_mngr.rs

```diff
@@ -51,6 +53,14 @@ pub fn sign_data(data: [u8; 32]) -> impl Future<Item = KeyedSignature, Error = f
     addr.send(Sign(data.to_vec())).flatten()
 }
 
+/// Sign a piece of (Hashable) data with the stored key.
+///
+/// This might fail if the manager has not been initialized with a key
+pub fn bls_sign(message: Vec<u8>) -> impl Future<Item = BlsKeyedSignature, Error = failure::Error> {
+    let addr = SignatureManagerAdapter::from_registry();
+    addr.send(BlsSign(message)).flatten()
+}
```

Comment:
Why `Vec<u8>`? It makes somehow hard to guess whether I should hash the bytes in advance or not.

Maybe we should make it uniform with how we deal with ECDSA: having separate `fn([u8; 32])` and `fn<T>(&T) where T: Hashable` functions.

Reply:
BLS signatures are implemented by hashing the message concatenated with a counter, like `H(message || counter)`. So if we use the hash instead of the message as the input, it would be hashing twice: `H(H(message) || counter)`. ECDSA works differently and can accept the message hash as input. So we can make the API uniform with ECDSA, but we will be hashing twice.